raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -748,18 +763,33 @@ contract TokenNetwork is Utils {
             require(token.transfer(partner, returned_tokens));
         }
 
-        emit ChannelUnlocked(participant, partner, computed_locksroot, unlocked_amount, returned_tokens);
-
-        // Channel must be settled and channel data deleted
-        require(channels[channel_identifier].state == 0);
```

Comment:
I removed this because the `channel_identifier` argument can be anything, including `0` or `> channel_counter`, making this check irrelevant.
`unlock` data discussion here: https://github.com/raiden-network/raiden-contracts/issues/202